### PR TITLE
[Hub Generated] Review request for Microsoft.AgFoodPlatform to add version preview/2020-05-12-preview

### PR DESCRIPTION
**Pull request by priyankajayaswal1**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/priyankajayaswal1/azure-rest-api-specs/dev-agfood-Microsoft.AgFoodPlatform-2020-05-12-preview...master/).

This PR is about merging AgFoodPlatform RP swagger.json for client sdk creation. We are approaching public preview next month and looking to initiate client creation process for the first time. We have already got approval for the changes mentioned in this PR in private repo (https://github.com/Azure/azure-rest-api-specs-pr/pull/3352).

I have used private repo commit as base in this PR  (https://github.com/Azure/azure-rest-api-specs/pull/13440/commits/77c5db25fe62c1ca5951728565035f809f389a33 and   https://github.com/Azure/azure-rest-api-specs/pull/13440/commits/6527fd86d2de8f5de5ead6d3c0d9349de36118e0 in this PR ) so that it is easy for you to review the changes appropriately. We want to merge this ASAP. Last commit contains only description updates.

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [x] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet. May 2021
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by sulabh-msft** [comment]
@Azure/arm-api-review-board Please review the PR.

**Comment by sulabh-msft** [comment]
@chiragg4u,
 Can you please review the PR ?

File: specification/agfood/resource-manager/Microsoft.AgFoodPlatform/preview/2020-05-12-preview/agfood.json [hunk 1]
```diff
@@ -0,0 +1,953 @@
...
+          }
+        },
+        "nextLink": {
+          "description": "Continuation link (absolute URI) to the next page of results in the list.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "FarmBeatsProperties": {
```
Comment (chiragg4u) [inline specification/agfood/resource-manager/Microsoft.AgFoodPlatform/preview/2020-05-12-preview/agfood.json:830]: >FarmBeatsProperties": [](http://example.com/codeflow?start=5&length=21)

Looks like there is no properties needs to be passed for creating a FarmBeats object. 
How are you corelating the backend instance with Azure instance?
Can customer link the existing instance to different Azure object?

**Comment by chiragg4u** [comment]
Overall model looks ok, I've minor non-blocking comments. Marking it ARMSignedoff, follow the process given @ https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/212/Swagger-PR-Review to complete and get the PR merged

File: specification/agfood/resource-manager/Microsoft.AgFoodPlatform/preview/2020-05-12-preview/agfood.json (hunk 1 above)
Comment (priyankajayaswal1) [reply specification/agfood/resource-manager/Microsoft.AgFoodPlatform/preview/2020-05-12-preview/agfood.json:830]: We use resourcename to create a dns zone entry and instance uri and this is used to map to the azure object. Hence. we don't need any farmbeatsProperties attribute for the mapping.

**Comment by priyankajayaswal1** [comment]
@chiragg4u thank you for your review. really appreciate it.
@jianyexi The last commit that you see I had unintentionally pushed which I reverted to keep it in sync with what Chirag approved.
image

Do let me know if there is anything else needed from our end to get this PR merged. 
Looking forward to you merging it ASAP.

File: specification/agfood/resource-manager/readme.go.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+## Go
+
+These settings apply only when `--go` is specified on the command line.
+
+```yaml $(go)
+go:
+  license-header: MICROSOFT_APACHE_NO_VERSION
```
Comment (ArcturusZhang) [inline specification/agfood/resource-manager/readme.go.md:7]: azure sdk for go is using MIT license now
```suggestion
  license-header: MICROSOFT_MIT_NO_VERSION
```

Comment (priyankajayaswal1) [reply specification/agfood/resource-manager/readme.go.md]: Updated

**Review by ArcturusZhang — COMMENTED** [review]
LGTM for the `readme.go.md`

File: specification/agfood/resource-manager/readme.python.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+``` yaml $(python-mode) == 'update' && $(track2)
+no-namespace-folders: true
+output-folder: $(python-sdks-folder)/agfood/azure-mgmt-agfood/azure/mgmt/agfood
+```
+
+``` yaml $(python-mode) == 'create' && $(track2)
+basic-setup-py: true
+output-folder: $(python-sdks-folder)/agfood/azure-mgmt-agfood
+```
```
Comment (msyyc) [inline specification/agfood/resource-manager/readme.python.md:48]:
```suggestion
## Python

These settings apply only when `--python` is specified on the command line.
Please also specify `--python-sdks-folder=<path to the root directory of your azure-sdk-for-python clone>`.

``` yaml $(track2)
azure-arm: true
license-header: MICROSOFT_MIT_NO_VERSION
package-name: azure-mgmt-agfood
no-namespace-folders: true
package-version: 1.0.0b1
clear-output-folder: true
```

``` yaml $(python-mode) == 'update' && $(track2)
no-namespace-folders: true
output-folder: $(python-sdks-folder)/agfood/azure-mgmt-agfood/azure/mgmt/agfood
```

``` yaml $(python-mode) == 'create' && $(track2)
basic-setup-py: true
output-folder: $(python-sdks-folder)/agfood/azure-mgmt-agfood
```
```

**Comment by msyyc** [comment]
@priyankajayaswal1 Please replace 'readme.python.md' with attachment :
[readme.python.md](https://github.com/Azure/azure-rest-api-specs/files/6439209/readme.python.md)

File: specification/agfood/resource-manager/readme.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+See configuration in [readme.typescript.md](./readme.typescript.md)
+
+## CSharp
+
+See configuration in [readme.csharp.md](./readme.csharp.md)
+
+## AzureResourceSchema
+
+See configuration in [readme.azureresourceschema.md](./readme.azureresourceschema.md)
+
```
Comment (jianyexi) [inline specification/agfood/resource-manager/readme.md:84]:
```suggestion
```

File: specification/agfood/resource-manager/readme.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+```yaml $(swagger-to-sdk)
+swagger-to-sdk:
+  - repo: azure-sdk-for-python-track2
+  - repo: azure-sdk-for-java
+  - repo: azure-sdk-for-go
+  - repo: azure-sdk-for-js
+  - repo: azure-resource-manager-schemas
+    after_scripts:
+      - node sdkauto_afterscript.js agfood/resource-manager
```
Comment (jianyexi) [inline specification/agfood/resource-manager/readme.md:58]:
```suggestion
```

File: specification/agfood/resource-manager/readme.azureresourceschema.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+## AzureResourceSchema
```
Comment (jianyexi) [inline specification/agfood/resource-manager/readme.azureresourceschema.md:1]: this file is unused, please remove it

File: specification/agfood/resource-manager/Microsoft.AgFoodPlatform/preview/2020-05-12-preview/agfood.json [hunk 7]
```diff
@@ -0,0 +1,953 @@
...
+          "Operations"
+        ],
+        "description": "Lists the available operations of Microsoft.AgFoodPlatform resource provider.",
+        "operationId": "Operations_List",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
```
Comment (jianyexi) [inline specification/agfood/resource-manager/Microsoft.AgFoodPlatform/preview/2020-05-12-preview/agfood.json:662]: Better to use ../../../../../common-types/resource-management/**v2**/types.json#/parameters/ApiVersionParameter

and the same with other places which refer to v1/types.json

File: specification/agfood/resource-manager/Microsoft.AgFoodPlatform/preview/2020-05-12-preview/agfood.json [hunk 8]
```diff
@@ -0,0 +1,953 @@
...
+          "readOnly": true
+        },
+        "extensionApiDocsLink": {
+          "description": "Extension api docs link.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ErrorResponse": {
```
Comment (jianyexi) [inline specification/agfood/resource-manager/Microsoft.AgFoodPlatform/preview/2020-05-12-preview/agfood.json:724]: Better to refer to v2/types.json#definitions/ErrorResponse

**Review by jianyexi — CHANGES_REQUESTED** [review]

File: specification/agfood/resource-manager/Microsoft.AgFoodPlatform/preview/2020-05-12-preview/agfood.json (hunk 8 above)
Comment (priyankajayaswal1) [reply specification/agfood/resource-manager/Microsoft.AgFoodPlatform/preview/2020-05-12-preview/agfood.json]: updated

**Review by jianyexi — APPROVED** [review]

**Pull request merged**